samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -100,8 +103,11 @@ public final void process(IncomingMessageEnvelope ime, MessageCollector collecto
   }
 
   @Override
-  public final void window(MessageCollector collector, TaskCoordinator coordinator) throws Exception {
-    // TODO: invoke timer based triggers
+  public final void window(MessageCollector collector, TaskCoordinator coordinator)  {
+    context.getSystemStreamPartitions().forEach(ssp -> {
+        OperatorImpl impl = this.operatorGraph.get(ssp.getSystemStream());
```

Comment:
Was mostly to prevent a long chained method invocation. (It makes it easier on my eyes to read two lines instead of one ;) ). 
Thanks for the suggestion. I've fixed it in the draft